internal/stores/contracts.go

```diff
@@ -287,3 +327,22 @@ func (s *SQLStore) contracts() ([]dbContract, error) {
 		Find(&contracts).Error
 	return contracts, err
 }
+
+func (s *SQLStore) AncestorContracts(id types.FileContractID, startHeight uint64) ([]bus.ArchivedContract, error) {
+	fcidGob, err := gobFCID(id)
+	if err != nil {
+		return nil, err
+	}
+	var ancestors []dbArchivedContract
+	err = s.db.Raw("WITH ancestors AS (SELECT * FROM archived_contracts WHERE renewed_to = ? UNION ALL SELECT archived_contracts.* FROM ancestors, archived_contracts WHERE archived_contracts.renewed_to = ancestors.fcid) SELECT * FROM ancestors WHERE start_height >= ?", fcidGob, startHeight).
```

Comment:
we should take into account the recursion limits here, it's configurable through `MAXRECURSION` hints and defaults to 100, which is already a high value but since it's tied to the period which is configurable perhaps we should add a hint here in function of the current height and start height? Or just max it out to 32767?

Reply:
I tried to max it out but it seems to only work in `SQLServer` and not `SQLite` ? Which has different limits and ways to set them? It would suck if there was no standard way to set it. If that is indeed the case maybe we should just revisit it later.

Reply:
I think we can pass `LIMIT -1` to the union query